src/schemaGenerator.ts

```diff
@@ -323,19 +327,39 @@ function addResolveFunctionsToSchema(schema: GraphQLSchema, resolveFunctions: IR
       }
       const field = fields[fieldName];
       const fieldResolve = resolveFunctions[typeName][fieldName];
+      const fieldValidator = (validations[typeName] && validations[typeName][fieldName]) || NOOP;
       if (typeof fieldResolve === 'function') {
         // for convenience. Allows shorter syntax in resolver definition file
-        setFieldProperties(field, { resolve: fieldResolve });
+        setFieldProperties(field, { resolve: validateThenResolve(fieldValidator, fieldResolve) });
       } else {
         if (typeof fieldResolve !== 'object') {
           throw new SchemaError(`Resolver ${typeName}.${fieldName} must be object or function`);
         }
-        setFieldProperties(field, fieldResolve);
+        setFieldProperties(field, validateThenResolve(fieldValidator, fieldResolve));
       }
     });
   });
 }
 
+function validateThenResolve(
+  validator: GraphQLFieldResolver<any, any>,
+  resolver: GraphQLFieldResolver<any, any>) {
+  return (
+    source: any,
+    args: { [argName: string]: any },
+    context: any,
+    info: GraphQLResolveInfo,
+  ) => Promise.resolve(validator(source, args, context, info))
+    .then(result => {
+      if (result) {
+        return resolver(source, args, context, info);
+      } else {
+        // TODO: Figure out how to report an error to the user here
```

Comment:
let's think about how we expect the behavior to be.
also, IMO we should also add some validator in type level so we can filter items in array
(for example, comments),
for standard fields, we need to think either we want to treat as null (will throw error on ! fields) or undefined